engine/config.cpp

```diff
@@ -0,0 +1,51 @@
+#include "config.h"
+
+#include "logger.h"
+
+#include <fstream>
+
+namespace pong
+{
+
+Config& Config::GetInstance()
+{
+    static Config instance;
+    return instance;
+}
+
+bool Config::LoadConfig(const std::string& filePath)
+{
+    std::ifstream jsonFile(filePath);
+    if (!jsonFile.is_open())
+    {
+        LogError("Failed to open JSON config file: {}", filePath);
+        return false;
+    }
+
+    nlohmann::json jsonData;
+
+    try
+    {
+        jsonFile >> jsonData;
+    }
+    catch (const nlohmann::json::parse_error& e)
+    {
+        LogError("JSON config parsing error: {}", e.what());
+        return false;
+    }
+
+    GetInstance().mConfig = jsonData;
+    return true;
+}
+
+std::optional<nlohmann::json> Config::GetJsonValue(const std::string& key)
```

Comment:
nice use of optional here. I am very infatuated with optional recently and it's ability to simplify interfaces

Reply:
Yeah, Im not using this function yet, but I had some ideas for some config values that may be different types. So I thought this would be the perfect use for it!